dvc/remote/local.py

```diff
@@ -92,6 +92,12 @@ def already_cached(self, path_info):
 
         return not self.changed_cache(current_md5)
 
+    def _verify_link(self, path_info, link_type):
+        if link_type == "hardlink" and self.getsize(path_info) == 0:
+            return
+
```

Comment:
Not great that it will be checking the size for each link it creates, might get expensive.

Reply:
We did that anyway on the `hardlink()` though.

Reply:
@skshetry Maybe there is some nicer way to do this? Like making `hardlink`(and other link class methods) verify themselves? This is an honest question, I don't know myself either :slightly_smiling_face:

Reply:
@skshetry Could keep it as is and create a ticket for it to reconsider later. Just asking.

Reply:
I like this suggestion https://github.com/iterative/dvc/pull/3428#issuecomment-596889084 more. Running verification on `hardlink` itself for only once (by setting `self.cache_type_confirmed`) is also hackish.

I'd say, we revisit this on next sprint and fix it?

Reply:
@skshetry Ok, please create an issue and add it to the next sprint.

Reply:
We could just check for `cache_type_verified` at the beggining again. I know that is duplication, but don't see any obvious workaround.